android/app/src/main/java/com/reactnativenavigation/controllers/NavigationActivity.java

```diff
@@ -133,6 +135,7 @@ protected void onNewIntent(Intent intent) {
     @Override
     protected void onPause() {
         super.onPause();
+        lastActivity = currentActivity;
```

Comment:
You're leaking the activity by saving it in a static field after `onPause`.

Note that `currentActivity` is static, `onDestroy` of first activity is called after `onResume` of second activity. Perhaps we can change the conditional in `destroyJsIfNeeded` to to:
`(currentActivity == null || (currentActivity.isFinishing() && currentActivity == this))` this way `onDestroyApp` is called only if the finishing activity is the current activity.

Reply:
Oh I see .. 

The problem is not related to the second part of the conditional but with the first one: `currentActivity == null` In fact, if the app is paused between the second screen `onResume` and the first screen `onDestroy` the `currentActivity` will be null and de app will be destroyed.

I will try to find another way to handle this case.